src/daemon/common.c

```diff
@@ -270,6 +270,8 @@ ssize_t swrite (int fd, const void *buf, size_t count)
 	ssize_t     status;
 	struct      pollfd pfd;
 
+	assert (fd >= 0);
```

Comment:
Imho, an assertion is not right. It's meant to catch programming errors in local code but not to validate arguments of exported functions.